src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
 
-      <Loader />
+  useEffect(() => {
+    const fetchUser = async () => {
+      if (currentTodo) {
+        const userFromServer = await getUser(currentTodo?.userId);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+        setUser(userFromServer);
+        setIsLoading(false);
+      }
+    };
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+    try {
+      fetchUser();
+    } catch {
+      setUser(null);
+    }
+  }, []);
```

Comment:
```suggestion
  }, [currentTodo]);
```
Just in case. It will save you against issues in future..